File: skillsphere-learning-service/src/main/java/com/infosis/nexus/learning/service/LearningPathService.java
package com.infosis.nexus.learning.service;

import com.infosis.nexus.learning.dto.LearningPathProgressResponse;
import com.infosis.nexus.learning.dto.LearningPathRequest;
import com.infosis.nexus.learning.dto.LearningPathResponse;

import java.util.List;

public interface LearningPathService {
    LearningPathResponse create(LearningPathRequest request);
    LearningPathResponse getById(Long id);
    List<LearningPathResponse> getAll();
    LearningPathResponse update(Long id, LearningPathRequest request);
    void delete(Long id);
    LearningPathProgressResponse getProgress(Long learningPathId, Long employeeId);
}

File: frontend/src/services/learningPathService.js
import api from './api';

const learningPathService = {
    getAll: () => api.get('/learning/learning-paths'),
    getById: (id) => api.get(`/learning/learning-paths/${id}`),
    create: (data) => api.post('/learning/learning-paths', data),
    update: (id, data) => api.put(`/learning/learning-paths/${id}`, data),
    delete: (id) => api.delete(`/learning/learning-paths/${id}`),
    getProgress: (id, employeeId) => api.get(`/learning/learning-paths/${id}/progress/${employeeId}`),
};

export default learningPathService;